jerry-core/ecma/base/ecma-gc.c

```diff
@@ -679,6 +688,18 @@ ecma_free_unused_memory (jmem_free_unused_memory_severity_t severity) /**< sever
   {
     JERRY_ASSERT (severity == JMEM_FREE_UNUSED_MEMORY_SEVERITY_HIGH);
 
+#ifndef CONFIG_ECMA_PROPERTY_HASHMAP_DISABLE
+    if (JERRY_CONTEXT (ecma_prop_hashmap_alloc_state) < ECMA_PROP_HASHMAP_ALLOC_MAX)
+    {
+      ++JERRY_CONTEXT (ecma_prop_hashmap_alloc_state);
+    }
+    if (JERRY_CONTEXT (ecma_prop_hashmap_alloc_last_is_hs_gc))
+    {
+      JERRY_CONTEXT (ecma_prop_hashmap_alloc_state) = ECMA_PROP_HASHMAP_ALLOC_MAX;
+    }
```

Comment:
I would reorder these to:

    if (JERRY_CONTEXT (ecma_prop_hashmap_alloc_last_is_hs_gc))
    {
      JERRY_CONTEXT (ecma_prop_hashmap_alloc_state) = ECMA_PROP_HASHMAP_ALLOC_MAX;
    }
    else if (JERRY_CONTEXT (ecma_prop_hashmap_alloc_state) < ECMA_PROP_HASHMAP_ALLOC_MAX)
    {
      ++JERRY_CONTEXT (ecma_prop_hashmap_alloc_state);
      JERRY_CONTEXT (ecma_prop_hashmap_alloc_last_is_hs_gc) = true;
    }